ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/FlexiblePrivacyGroupContract.java

```diff
@@ -229,12 +228,13 @@ private CallParameter buildCallParams(final Bytes methodCall) {
         Address.ZERO, FLEXIBLE_PRIVACY_PROXY, 3000000, Wei.of(1000), Wei.ZERO, methodCall);
   }
 
-  private List<String> decodeList(final Bytes rlpEncodedList) {
+  public static List<String> decodeList(final Bytes rlpEncodedList) {
     final ArrayList<String> decodedElements = new ArrayList<>();
     // first 32 bytes is dynamic list offset
     if (rlpEncodedList.size() < 64) return decodedElements;
-    final long lengthOfList =
-        UInt256.fromBytes(rlpEncodedList.slice(32, 32)).toLong(); // length of list
+    // Bytes uses a byte[] for the content which can only have up to Integer.MAX_VALUE-5 elements
+    final int lengthOfList =
+        UInt256.fromBytes(rlpEncodedList.slice(32, 32)).toInt(); // length of list
```

Comment:
nit: please remove the UInt256.fromBytes, as you can do the toInt() on the slice (Bytes).

Reply:
as discussed, this needs to stay :)

Reply:
Value of size 32 has more than 4 bytes
Without this, running tests I get this error: 
java.lang.IllegalArgumentException: Value of size 32 has more than 4 bytes
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:193)
	at org.apache.tuweni.bytes.Bytes.toInt(Bytes.java:685)
	at org.apache.tuweni.bytes.Bytes.toInt(Bytes.java:673)
	at org.hyperledger.besu.ethereum.privacy.FlexiblePrivacyGroupContract.decodeList(FlexiblePrivacyGroupContract.java:236)